### PR TITLE
textarea mixin

### PR DESCRIPTION
**Pull request by easternbloc**
This adds support for a textarea mixin.

I've removed attributes that don't work with Input but for ease I've reused the inputText method as most things are the same.

### CODE REVIEW

File: partials/forms/textarea-group.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        {{{label}}}
+        {{#hint}}<span {{$hintId}}id="{{hintId}}" {{/hintId}}class="form-hint">{{hint}}</span>{{/hint}}
+        {{#error}}<span class="visuallyhidden">{{error.message}}</span>{{/error}}
+    </label>
+    <textarea
+        name="{{id}}"
+        id="{{id}}"
+        class="form-control{{#className}} {{className}}{{/className}}{{#error}} invalid-input{{/error}}"
+        aria-required="{{required}}"
+        {{#maxlength}} maxlength="{{maxlength}}"{{/maxlength}}
```
Comment (lennym) [inline partials/forms/textarea-group.html]: `maxlength` for textareas is not a thing.

Comment (easternbloc) [inline partials/forms/textarea-group.html]: except that it is: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea

Comment (lennym) [inline partials/forms/textarea-group.html]: Well... that's new.

Comment (lennym) [inline partials/forms/textarea-group.html]: * probably not that new

Comment (easternbloc) [inline partials/forms/textarea-group.html]: upon winning an argument with Lenny

**Comment by lennym** [comment]
I feel like the level of duplication between the text area template and the input text template warrants some level of re-use.

**Comment by lennym** [comment]
Also, following conversation with @gavboulton we think that the `compound` stuff is perhaps wildly overengineered, and so we may remove it. But that's another thing.

Do you use the `compound` stuff?

**Comment by easternbloc** [comment]
I thought that but then I thought we don't gain very much doing that and it creates more overhead in understanding what's going on. I think the template should be quick to understand and modify. This gives us more flexibility all be it with a bit more template code.

**Comment by easternbloc** [comment]
At the moment we don't use any `compound` stuff no.

**Comment by easternbloc** [comment]
I left it in with the textarea just so it plays nicely with everything else.

**Comment by lennym** [comment]
No, I just had a look through various things to work out what it did, and it literally just ends up setting a different bottom margin, and since we already have the facility to define `className`s then it probably doesn't warrant the overhead of its own functions.

**Comment by easternbloc** [comment]
What would you like me to do then as I'd rly like to get this merged as it's linked to a story I want to be able to PR later today

**Comment by lennym** [comment]
Not necessarily for this PR (unless @gavboulton has strong feelings), but maybe *at some point in the future*.

**Comment by gavboulton** [comment]
It's fine with me to merge.

**Comment by lennym** [comment]
Published @0.6.0

**Comment by easternbloc** [comment]
:+1:

**Pull request merged**